synapse/handlers/auth.py

```
@@ -452,40 +461,165 @@ def _check_local_password(self, user_id, password):
 
     @defer.inlineCallbacks
     def _check_ldap_password(self, user_id, password):
-        if not self.ldap_enabled:
-            logger.debug("LDAP not configured")
+        """ Attempt to authenticate a user against an LDAP Server
+            and register an account if none exists.
+
+            Returns:
+                True if authentication against LDAP was successful
+        """
+
+        if not ldap3 or not self.ldap_enabled:
             defer.returnValue(False)
 
-        import ldap
+        if self.ldap_mode not in ('simple_bind', 'search'):
+            logger.warn('Invalid ldap mode specified: %s', self.ldap_mode)
```

I would raise an exception here, given its synapse that sets the `ldap_mode`